src/modelfree/scheduling.py

```diff
@@ -63,8 +95,9 @@ def get_value(self, step):
 
 class ConstantAnnealer(Annealer):
```

Comment:
`ConstantAnnealer` and `LinearAnnealer` don't seem to use `shaping_env` -- why is it an argument to the constructor? Consider removing.

Also, both of these classes are missing `set_shaping_env`, but you seem to assume in `set_annealer_shaping_env` that all `Annealer`'s have this method. Please modify the `Annealer` class to have a default implementation of `set_shaping_env`. This could set `self.shaping_env`, throw a `NotImplementedError`, or just be a no-op -- but it must exist!